### PR TITLE
Make `options` an optional paramter

### PR DESCRIPTION
**Pull request by izolate**

### CODE REVIEW

File: lib/millify.ts [hunk 1]
```diff
@@ -35,19 +35,19 @@ function* divider(value: number, base: number): IterableIterator<number> {
  * @param {number} value - Number to convert
  * @param {Options} options
  */
-function Millify(value: number, options: Options): string {
+function Millify(value: number, options = {}): string {
   // Override default options with supplied ones
   const opts: Options = { ...defaultOptions, ...options };
```
Comment (amio) [inline lib/millify.ts]: How about this? We can preserve `Options` interface in the signature:

```js
function Millify(value: number, options?: Options): string {
  // Override default options with supplied ones
  const opts: Options = { ...defaultOptions, ...options || {} };
```

Comment (amio) [reply lib/millify.ts]: or

```js
function Millify(value: number, options?: Options): string {
  // Override default options with supplied ones
  const opts: Options = Object.assign({}, defaultOptions, options);
```

Comment (izolate) [reply lib/millify.ts]: Agree that preserving the `Options` interface in the parameter is preferred. The problem is that now users have to supply the *entire* options object:

```typescript
import millify from 'millify';

millify(1000); // OK

millify(1000, { precision: 2 });

error TS2345: Argument of type '{ precision: number; }' is not assignable to parameter of type 'Options'.
  Type '{ precision: number; }' is missing the following properties from type 'Options': decimalSeparator, lowerCase, space, base, units
```

I'm looking for a solution that:

1) Preserves the `Options` interface in the parameter
2) Allows users to update a single option
3) Doesn't require adding optional attributes to the `Options` interface

I proposed a `UserOptions` interface [here](https://github.com/izolate/millify/pull/14/commits/7d1820716ef1ce9f251bdc702deb3f5bb5afc977). What do you think?

Comment (amio) [reply lib/millify.ts]: > The problem is that now users have to supply the entire options object

Forgot that 😂

+1 on `UserOptions`, we can use [`Partial<T>`](https://typescript-play.js.org/#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgeTBgQiQGc4BvAKDjjCmGwTJKQC44kBXAWwCM0AblpwAJkwS9MAGwDKwMJiiYY0TmRhRkAcxF0ZEAO5oAwpjLBO-CBBnBMSfXDJLc12-cfP+Fq1z5BKGduJAQYMg0tXQBtAF0RAF9qalBIWDgYAE8wPABVSyhCYlIKAF44AAVlYlkAHmK2MgA+FOxS+AgiTgK0RtK4Cpo6OgZJVlJOAGZqRKA) for this case.

Comment (izolate) [reply lib/millify.ts]: > we can use Partial<T> for this case.

Genius. Thanks for the tip! Did require the use of non-null assertions (`!`) but aside from that it works great.

Comment (izolate) [reply lib/millify.ts]: Released in [v3.1.3](https://github.com/izolate/millify/releases/tag/v3.1.3). Thanks for your help @amio

**Pull request merged**